File: src/Self_defined_Msg/scripts/main_talker.py
#! /usr/bin/env python 
# coding:utf-8

import  rospy
from Self_defined_Msg.msg import person


if __name__ == "__main__":
    rospy.init_node("talker")
    pub = rospy.Publisher("chat",person,queue_size=10)
    r = rospy.Rate(10)

    p = person()
    p.name = "name"
    p.age = 20
    p.height = 166.6

    rospy.sleep(rospy.Duration(1))   # 在C++裡面是這樣打的: ros::Duration(1).sleep()
    while not rospy.is_shutdown():
        rospy.loginfo("Talker publish: %s, %d, %f",p.name, p.age, p.height)
        pub.publish(p)
        p.age += 1
        r.sleep()
        

File: src/Self_defined_Msg/src/main_talker.cpp
# include "ros/ros.h"
# include "person.h"
# include "sys/time.h"


int main(int argc,char*argv[]){
    ros::init(argc,argv,"talker");
    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<Self_defined_Msg::person>("chat",10);
    
    Self_defined_Msg::person p;
    p.name = "Name";
    p.age = 20;
    p.height = 170;

    ros::Rate r(10);
    ros::Duration(1).sleep();

    while(ros::ok()){
        ROS_INFO("Talker Publish: %s, %d, %f",p.name.c_str(), p.age, p.height);
        pub.publish(p);
        p.age++;
        r.sleep();
        ros::spinOnce();
    }

    return 0;

}